lib/singlepass-backend/src/codegen_x64.rs

```diff
@@ -1536,27 +1538,107 @@ impl X64FunctionCode {
             a.emit_add(Size::S64, Location::GPR(tmp_base), Location::GPR(tmp_addr));
             a.emit_cmp(Size::S64, Location::GPR(tmp_bound), Location::GPR(tmp_addr));
             a.emit_conditional_trap(Condition::Above);
-        }
 
-        m.release_temp_gpr(tmp_bound);
+            m.release_temp_gpr(tmp_bound);
+        }
 
         // Calculates the real address, and loads from it.
         a.emit_mov(Size::S32, addr, Location::GPR(tmp_addr));
-        if offset != 0 {
+        if memarg.offset != 0 {
             a.emit_add(
                 Size::S64,
-                Location::Imm32(offset as u32),
+                Location::Imm32(memarg.offset as u32),
                 Location::GPR(tmp_addr),
             );
         }
         a.emit_add(Size::S64, Location::GPR(tmp_base), Location::GPR(tmp_addr));
         m.release_temp_gpr(tmp_base);
 
+        let align = match memarg.flags & 3 {
+            0 => 1,
+            1 => 2,
+            2 => 4,
+            3 => 8,
+            _ => unreachable!("this match is fully covered"),
+        };
+        if check_alignment && align != 1 {
+            let tmp_aligncheck = m.acquire_temp_gpr().unwrap();
+            //let tmp_mask = m.acquire_temp_gpr().unwrap();
+            a.emit_mov(
+                Size::S32,
+                Location::GPR(tmp_addr),
+                Location::GPR(tmp_aligncheck),
+            );
+            //a.emit_mov(Size::S64, Location::Imm64(align - 1), Location::GPR(tmp_mask));
+            //a.emit_and(Size::S64, Location::GPR(tmp_mask), Location::GPR(tmp_aligncheck));
```

Comment:
Is this debug code?  Probably best to remove it or explain in a comment why it's there

Reply:
Deleted.

It's the old code. We're under heavy register pressure here, and I rewrote it to use one less register, but kept the original around in case my changes didn't work properly. Thanks!